### PR TITLE
Add vagrant directory helper

### PR DESCRIPTION
**Pull request by deizel**
Auto-`cd` to `/vagrant` when you `vagrant ssh` into box and the dotfiles load.

### CODE REVIEW

File: zshrc.d/10-vagrant [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Vagrant-specific settings
+
+# Jump to mounted working directory rather than $HOME
+if [[ $(pwd) == "/home/vagrant" ]]; then
+  cd /vagrant
+fi
```
Comment (jadb) [inline zshrc.d/10-vagrant]: Isn't there a way to check also that one is on a VM? While it's rare to have `/home/vagrant` on a local machine, it isn't impossible so maybe an extra check (if one exists) would be good.

Comment (deizel) [inline zshrc.d/10-vagrant]: You mean there are actually people out there with the username `vagrant`? :stuck_out_tongue:

**Comment by deizel** [comment]
Not sure if this is the best way to detect if we are inside a Vagrant VM.

It works, but let me know if you have a better idea.

**Comment by jadb** [comment]
I don't know of any myself but I am sure (or at least hopeful) there is something cleaner than that ;)

Let's keep it open and see if we find anything good in the next couple days.

**Comment by jadb** [comment]
Merging as it doesn't look like we had any interest in further improving it.

**Pull request merged**